Sources/Swift/Integrations/UserFeedback/SentryUserFeedbackFormViewModel.swift

```diff
@@ -510,77 +224,278 @@ class SentryUserFeedbackForm: UIViewController {
+        emailTextFieldHeightConstraint.constant = formElementHeight * config.scaleFactor
+        addScreenshotButtonHeightConstraint.constant = formElementHeight * config.scaleFactor
+        removeScreenshotButtonHeightConstraint.constant = formElementHeight * config.scaleFactor
+        submitButtonHeightConstraint.constant = formElementHeight * config.scaleFactor
+        cancelButtonHeightConstraint.constant = formElementHeight * config.scaleFactor
+    }
+    
+    func updateScreenshot(image: UIImage, accessibilityInfo value: String) {
         screenshotImageView.image = image
-        screenshotImageAspectRatioConstraint.isActive = false
-        screenshotImageAspectRatioConstraint = screenshotImageView.widthAnchor.constraint(equalTo: screenshotImageView.heightAnchor, multiplier: image.size.width / image.size.height)
-        screenshotImageAspectRatioConstraint.isActive = true
+        screenshotImageView.accessibilityLabel = value
         addScreenshotButton.isHidden = true
         removeScreenshotStack.isHidden = false
+        
+        let aspectRatio: CGFloat
+        if image.size.height == 0 {
+            #if !SENTRY_TEST
+            SentryLog.warning("Image had 0 height, won't be able to set a reasonable aspect ratio. Defaulting to 1:1.")
+            #endif // !SENTRY_TEST
+            aspectRatio = 1
+        } else {
+            aspectRatio = image.size.width / image.size.height
+        }
+        
+        // you cannot dynamically change the multiplier on a constraint. you must deactivate the old instance, create a new instance, and then activate that one
+        screenshotImageAspectRatioConstraint.isActive = false
+        screenshotImageAspectRatioConstraint = screenshotImageView.widthAnchor.constraint(equalTo: screenshotImageView.heightAnchor, multiplier: aspectRatio)
+        screenshotImageAspectRatioConstraint.isActive = true
+        
+        updateSubmitButtonAccessibilityHint()
+    }
+    
+    typealias SentryUserFeedbackFormValidation = Result<String, InputError>
+    func validate() -> SentryUserFeedbackFormValidation {
+        var missing = [String]()
+        var hint = ["Will submit feedback"]
+        
+        if let name = fullNameTextField.textOrNil {
+            hint.append("for \(name)")
+        } else if config.formConfig.isNameRequired {
+            missing.append(config.formConfig.nameLabel.lowercased())
+        } else {
+            hint.append("with no name")
+        }
+        
+        if let email = emailTextField.textOrNil {
+            hint.append("at \(email)")
+        } else if config.formConfig.isEmailRequired {
+            missing.append(config.formConfig.emailLabel.lowercased())
+        } else {
+            if fullNameTextField.hasText {
+                hint.append("with no email address")
+            } else {
+                hint.append("or email address")
+            }
+        }
```

Comment:
```suggestion
        } else if fullNameTextField.hasText {
            hint.append("with no email address")
        } else {
            hint.append("or email address")
        }
```

Reply:
I iterated a lot on this part and feel like at some point I couldn't do this, but it makes sense now, so i'll take your suggestion. Thanks!

Reply:
Thanks, this part went through a lot of iterations and didn't seem like i could do this at some point.